network-api/networkapi/campaign/views.py

```diff
@@ -52,10 +52,15 @@ def send_message(self, QueueUrl, MessageBody):
 logger = logging.getLogger(__name__)
 
 
-@api_view(['POST'])
-@parser_classes((JSONParser,))
-@permission_classes((permissions.AllowAny,))
+@csrf_exempt
+@require_http_methods(['POST'])
 def signup_submission_view(request, pk):
+    # We need to re-write the data that's coming in from the XMLHttpRequest.
+    # XMLHttpRequest's send data through the request.body, not request.POST despite it being a POST method
+    # request.POST is supported for unit tests
+    new_body = request.body.decode("utf-8")
+    request.data = json.loads(new_body)
```

Comment:
Note that `json.loads` can throw a `ValueError`, so we'll want to try/except this.

Reply:
Good catch!